roadmap/implementers-guide/src/runtime/disputes.md

```diff
@@ -17,90 +19,351 @@ We account for these requirements by having the validity module handle two kinds
+## Remote Disputes
 
-> TODO: store all included candidate and attestations on them here. accept additional backing after the fact. accept reports based on VRF. candidate included in session S should only be reported on by validator keys from session S. trigger slashing. probably only slash for session S even if the report was submitted in session S+k because it is hard to unify identity
+On dispute detection, the full block that is disputed is gossiped to all peers.
 
-One first question is to ask why different logic for local disputes is necessary. It seems that local disputes are necessary in order to create the first escalation that leads to block producers abandoning the chain and making remote disputes possible.
+Votes in favour of block validity always includes the primary votes from the `Backing` stage.
 
-Local disputes are only allowed on parablocks that have been included on the local chain and are in the acceptance period.
+Votes aginst the block validity always include the initial challenger from the `Backing` stage iff any.
 
-For each such parablock, it is guaranteed by the inclusion pipeline that the parablock is available and the relevant validation code is available.
+Session change implies a change of session keys. Hence at least two validator sets must be kept - the set of validators in the current session and the ones in the previous session.
+The timeouts must be set accordingly to guarantee that two sessions are sufficient, and timeout actions are not triggered in a 3rd session.
+An era change implies a session change, since the nominators change.
 
-Disputes may occur against blocks that have happened in the session prior to the current one, from the perspective of the chain. In this case, the prior validator set is responsible for handling the dispute and to do so with their keys from the last session. This means that validator duty actually extends 1 session beyond leaving the validator set.
+When a dispute has occurred on another fork, we need to transplant that dispute to every other fork. This poses some major challenges.
 
-...
+There are two types of remote disputes. **Concluded** and **Unconcluded**.
 
-After concluding with enough validtors voting, the dispute will remain open for some time in order to collect further evidence of misbehaving validators, and then issue a signal in the header-chain that this fork should be abandoned along with the hash of the last ancestor before inclusion, which the chain should be reverted to, along with information about the invalid block that should be used to blacklist it from being included.
+The first is a remote roll-up of a **concluded dispute**. These are simply all attestations for the block, those against it, and the result of all (secondary) approval checks. A concluded remote dispute can be resolved in a single transaction as it is an open-and-shut case of a quorum of validators disagreeing with another.
```

Comment:
This is true, but not sure if this is the easiest way to implement it.  It's maybe easier to replay everything in some dispute context.  We can afford to do that since someone is being slashed 100%.